gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java

```diff
@@ -266,8 +278,29 @@ public void showSettings() {
     navigator.navigate(HomeScreenFragmentDirections.actionHomeScreenFragmentToSettingsActivity());
   }
 
-  public void onGeoJsonClick(MapGeoJson mapGeoJson) {
-    showBottomSheet(mapGeoJson.getFeature());
+  public Observable<ImmutableList<Feature>> getOverlappingFeatures() {
+    return candidateFeatures;
+  }
+
+  public void onFeatureClick(ImmutableList<MapFeature> mapFeatures) {
+    showFeatureSelector(mapFeatures);
+  }
+
+  private void showFeatureSelector(ImmutableList<MapFeature> mapFeatures) {
+    ImmutableList<Feature> features =
+        stream(mapFeatures)
+            .map(
+                feature -> {
+                  if (feature instanceof MapPin) {
+                    return ((MapPin) feature).getFeature();
+                  } else if (feature instanceof MapGeoJson) {
+                    return ((MapGeoJson) feature).getFeature();
+                  }
+                  return null;
+                })
+            .filter(f -> f != null)
+            .collect(toImmutableList());
+    candidateFeatures.onNext(features);
```

Comment:
Since `candidateFeatures` is subscribed to by the same class, can we just get rid of it and call `onTapOverlappingFeatures` directly?

Reply:
The `Fragment` subscribes to it, not the view model. We could still remove it, but this would entail moving the locus of control over UI from the fragment into the VM, which I think we should avoid. This approach is also consistent with the way we pass other information across `map container -> home VM -> home fragement`

Reply:
However, I will use `LiveData`